pkg/certificate/reconciler.go

```diff
@@ -0,0 +1,51 @@
+package certificate
+
+import (
+	"context"
+	"time"
+
+	"github.com/openservicemesh/osm/pkg/errcode"
+)
+
+// CheckAndUpdate checks for a MRC to reach a desired condition and when the condition is met,
+// it updates the status and terminates the loop
+func (r *MRCReconciler) CheckAndUpdate(ctx context.Context, mrcClient MRCClient, checkInterval time.Duration) {
+	ticker := time.NewTicker(checkInterval)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				ticker.Stop()
+				log.Info().Msgf("Stopping status reconciliation loop for MRC %s", r.mrcName)
+				return
+			case <-ticker.C:
+				mrc := mrcClient.GetMeshRootCertificate(r.mrcName)
+				if mrc == nil {
+					log.Error().Msgf("failed to get MRC %s in status reconciliation loop", r.mrcName)
+					continue
+				}
+
+				meetsCondition, err := r.checkStatus(ctx, mrc)
+				// checkStatus will return an error if the MRC is no longer in the expected rotation state
+				// i.e. prior to this reconciliation loop, another component has already moved the rotation
+				// to the next state after seeing the valid status condition without this check, the MRC
+				// reconciliation loop might never terminate
+				if err != nil {
```

Comment:
I'd like to see the ability to change behavior (e.g. log, emit a metric, etc.) based on the type of error. For the situation described in the comment (MRC is no longer in the expected rotation state), could you emit a specific error type (e.g. `ErrUnexpectedMRCStatusInReconciler`)?